File: test-artifacts/p-visitor-py/function-invocation-replacer/int88888888_044_in.py
def f_gold(first, second):
    if len(first) == 0 and len(second) == 0:
        return True
    if len(first) > 1 and first[0] == "*" and len(second) == 0:
        return False
    if (len(first) > 1 and first[0] == "?") or (len(first) != 0 and len(second) != 0 and first[0] == second[0]):
        retval_1 = f_gold(first[1:], second[1:])
        return retval_1
    if len(first) != 0 and first[0] == "*":
        retval_2 = f_gold(first[1:], second) or f_gold(first, second[1:])
        return retval_2
    return False

File: test-artifacts/p-visitor-js/function-invocation-replacer/int88888888_044_in.js
function f_gold(first, second) {
    if (first.length === 0 && second.length === 0) {
        return true;
    }
    if (first.length > 1 && first[0] === "*" && second.length === 0) {
        return false;
    }
    if ((first.length > 1 && first[0] === "?") || (first.length !== 0 && second.length !== 0 && first[0] === second[0])) {
        const retval_1 = f_gold(first.slice(1), second.slice(1));
        return retval_1;
    }
    if (first.length !== 0 && first[0] === "*") {
        const retval_2 = f_gold(first.slice(1), second) || f_gold(first, second.slice(1));
        return retval_2;
    }
    return false;
}